### DISCUSSION TITLE
Version Control

### DISCUSSION THREAD

**Discussion by Kaedriz** [soulhotel/FF-ULTIMA#12] [General]
Is there any particular reason you picked this file versioning policy? Like separate folders for different releases defeats the purpose of Git, I suppose? Maybe for less technical people this would be easier to download earlier versions, but this theme, as of now, is best consumed in the newest format, it is backward compatible, so no reason to download earlier versions.

Another problem that stems from above, albeit maybe this is only bugging me. Wouldn't structuring a theme in a way that is drop in better for consumers to update? Like I had an idea, users would just clone this repo in chrome directory, and occasionally pull updates if they were any. This would also enable easy customization in the form of forks. Like, if you wanted slightly different values for something, you would just fork and change it in your repo. This would enable easy updates for these customizing things. Occasional merge issues could arise, but they would be pretty easy to resolve for anyone inclined. I theoretically(?) could accomplish this by symlinks, but I think this would be a great addition to QoL of usage of this theme.

I am curious about your though on this topic.

**Comment by soulhotel** [comment]
Really good questions. Im very much aware that I may not be providing updates in the easiest manner, probably because i'm also _unaware_ of everything I can do with github. (started using it 3-4 months)

And Im assuming youre talking about how ive uploaded 1.5, 1.6, 1.6.5, folders to the main branch, yeah? Would keeping it as just the latest version be better, and for updates i would use vscode to 'commit'?

Also tell me more about how you would implement symlinks

**Reply by Kaedriz** [reply]
In the context of symlinks, I just gave them as mine resolution to this problem, essentially just git clone this project somewhere else, and link it to Firefox chrome folder, but probably just doable on Linux, maybe mac.

Yea, I think just keeping latest version would be better, as you can preview earlier version on GitHub. Essentially this would become main folder + maybe some README Markdown and other
![Screenshot_20240404_005346](https://github.com/soulhotel/FF-ULTIMA/assets/44523955/60ec9830-6101-4354-86dc-2202d3964e75)

I, myself, am not an expert on GitHub, so can be confusing things.

Here, you already started using releases/tags. I suppose they are used for version tagging. Clicking them moves you to a specific state of repo when that tag was committed?
![Screenshot_20240404_005635](https://github.com/soulhotel/FF-ULTIMA/assets/44523955/b0273f13-5b05-4659-b04c-1546947ac58d)

This would also avoid polluting folder structure when more releases arrive.

Only problem of this solution is that previous commits would be delinked from these files, as they have changed location, but don't quote me on that, need to check that if git is smart enough.

UPDATE: Ye, I checked, and git history is moved, so even that is no issue.

**Reply by Kaedriz** [reply]
It would look something like [this](https://github.com/KaCuQ/FF-ULTIMA).
All you would need to do as an end user is to git clone this repo to your chrome folder, or just simply download zip and extract there.

EDIT: Aside from this, there would also be a necessary change to Docs to update their instructions. I can try to write a guide to setting up git way for autoupdating via git.

**Reply by soulhotel** [reply]
Even if the history was removed, it still better to structure now I would say. So Thank You for the recommendation. As for the tags youre correct, they already let users navigate to previous releases. So.. I made changes.

- The source is structured according to the chrome folder
- the modification page, and replace features pages are redirected to the chrome/help/ folder
- included in the help folder is a preview folder for images, so every guide/tutorial for modifications/etc can be viewed easily in a chosen text editor even offline
- so from now on when i make an updated release i just commit changes with vscode, then create a new release.

![Screenshot_1](https://github.com/soulhotel/FF-ULTIMA/assets/155501797/32bdad0b-3491-4f44-9d1e-75016c074a7a)

**Reply by Kaedriz** [reply]
You could also just create new folder "modifications" or "optional" etc. to hold all optional stuff instead of help. And in "help" only hold some Markdown guides etc. It would be less confusing, maybe.

**Reply by Kaedriz** [reply]
Oh, and another one being a base indentation of some sort. I feel it would really help to set a baseline formation to ease in contributing. Not even .editorconf, just simple making sure to have all files in the same format. My personal recommendation would be Tabs + 2/4 units, but that's your decision.
